test/tools/cli/IonJavaCli.java

```diff
@@ -0,0 +1,1341 @@
+            String importName = null;
+            int maxId = 0;
+            int version = 0;
+
+            ionReader.stepIn();
+            while (ionReader.next() != null) {
+                switch (ionReader.getFieldName()) {
+                    case "name":
+                        importName = ionReader.stringValue();
+                        break;
+                    case "max_id":
+                        maxId = ionReader.intValue();
+                        break;
+                    case "version":
+                        version = ionReader.intValue();
+                        break;
+                }
+            }
+            ionReader.stepOut();
+
+            ImportDescriptor table = new ImportDescriptor(importName, maxId, version);
+            imports.add(table);
+        }
+        ImportDescriptor[] importsArray = imports.toArray(new ImportDescriptor[0]);
+
+        ionReader.stepOut();
+        return importsArray;
+    }
+
+    /**
+     *  This function creates a new file for output stream with the fileName, If file name is empty or undefined,
+     *  it will create a default output which is System.out or System.err.
+     *
+     *  this function never return null
+     */
+    private static BufferedOutputStream initOutputStream(CommandArgs args,
+                                                         String defaultValue,
+                                                         ProcessContext processContext) throws IOException {
+        BufferedOutputStream outputStream = null;
+        switch (defaultValue) {
+            case SYSTEM_OUT_DEFAULT_VALUE:
+                String outputFile = args.getOutputFile();
+                if (outputFile != null && outputFile.length() != 0) {
+                    File myFile = new File(outputFile);
+                    if (args.getCommand() == CommandType.PROCESS && processContext != null) {
+                        processContext.setFile(myFile);
+                    }
+                    FileOutputStream out = new FileOutputStream(myFile);
+                    outputStream = new BufferedOutputStream(out, BUFFER_SIZE);
+                } else {
+                    outputStream = new BufferedOutputStream(System.out, BUFFER_SIZE);
+                }
+                break;
+            case SYSTEM_ERR_DEFAULT_VALUE:
+                String errorReport = args.getErrorReport();
+                if (errorReport != null && errorReport.length() != 0) {
+                    File myFile = new File(errorReport);
+                    FileOutputStream out = new FileOutputStream(myFile, true);
+                    outputStream = new BufferedOutputStream(out, BUFFER_SIZE);
+                } else {
+                    outputStream = new BufferedOutputStream(System.err, BUFFER_SIZE);
+                }
+                break;
+        }
+        return outputStream;
+    }
+
+    static class Pair {
+        final int fst;
+        final int snd;
```

Comment:
Now that we have control over this class, let's give these better names. Maybe `left` and `right` to match the variable names you use to construct them.